### PR TITLE
Refactor all React class components to hooks

### PR DESCRIPTION
**Pull request by pyxld-kris**
#9

### CODE REVIEW

**Comment by pyxld-kris** [comment]
A few extra commits in there dealing with, and then removing, non-applicable code. I accidentally branched from an in-progress dev branch ( #5 )

Only commit with changes should be in 'Refactor all class components to use hooks' (commit 2630fca010d912f708f6e9e31fc8670149c57096)

File: src/routes/PlayRoute.js [hunk 1]
```diff
@@ -7,16 +7,14 @@ import Header from "../components/modules/Header";
...
-  }
+export default function PlayRoute() {
+  const [projects, setProjects] = React.useState([]);
+
+  React.useEffect(() => {
+    populateProjects();
+  }, []);
 
-  populateProjects() {
+  const populateProjects = () => {
```
Comment (third774) [inline src/routes/PlayRoute.js]: It looks like the body of this function could just be moved into the `useEffect` body?

**Comment by pyxld-kris** [comment]
Love having the expert insight @third774 !

Will be going through and modifying this now using some of your input. I'm also looking to get some opinions on other things in the project: structure, UI framework, Redux vs Context API, etc. Hoping you'll weigh in on some of the issues tied to those as well!

File: src/components/common/Button/Button.js [hunk 2]
```diff
@@ -6,29 +6,22 @@ import { toast } from "react-toastify";
...
-        className={style.Button}
-        style={inlineStyles}
-        onClick={this.props.onClick ? this.props.onClick : this.clickHandler}
-      >
-        {this.props.children}
-      </button>
-    );
-  }
+  let inlineStyles = {};
+  if (props.fontSize) inlineStyles = { fontSize: props.fontSize };
```
[the remark this answers is not in the document]
Comment (pyxld-kris) [reply src/components/common/Button/Button.js]: Implemented and seems to be working. Commit f138e02f47fb425e7637801b42b3ab4a638d160f

File: src/components/modules/Footer/Footer.js [hunk 3]
```diff
@@ -5,45 +5,8 @@ import { Link } from "react-router-dom";
...
-        <div className={style.OrgInfoArea}>
-          Terms of Service | Privacy Policy | ©Dev Launchers, 2020.
-        </div>
-      </footer>
-    );
-  }
-
-  getRandomQuote() {
+export default function Footer(props) {
+  const getRandomQuote = () => {
```
[the remark this answers is not in the document]
Comment (pyxld-kris) [reply src/components/modules/Footer/Footer.js]: Moved quotes to content layer and created RandomQuote component in Footer. Commit 40ea2a1967e1afc317e670924121f88ee0632ed4

I do wonder if this is acceptable design moving forward, or if there is a better place to put this sort of logic. Specifically anticipating that we'll be moving all of our content out of local files in the future with content being served up from some sort of API. I considered creating a useRandomQuote hook, and potentially stuffing it into the 'utils' folder for now, or creating a top level 'hooks' folder to put common custom hooks into (Cursory search for hook structure turned this up: https://www.relevantprogrammer.com/posts/todo-with-react-hooks/ and also in answers here https://www.reddit.com/r/reactjs/comments/b9puyb/where_are_people_organizing_their_hooks/)

If you have any thoughts or resources about project structure, they could be invaluable in making sure we're structuring things in a scalable way that will make sense to everyone.

File: src/components/modules/Footer/Footer.js [hunk 4]
```diff
@@ -53,16 +16,37 @@ export default class Footer extends React.Component {
...
+        </Link>
+        <Link to={"/events"} className="nav-link">
+          <div className={style.NavEntry}>EVENTS</div>
+        </Link>
+        <Link to={"/support-us"} className="nav-link">
+          <div className={style.NavEntry}>SUPPORT US</div>
+        </Link>
+      </div>
+      <div className={style.OrgInfoArea}>
+        Terms of Service | Privacy Policy | ©Dev Launchers, 2020.
```
[the remark this answers is not in the document]
Comment (pyxld-kris) [reply src/components/modules/Footer/Footer.js]: Will create an issue for this!

File: src/routes/EarnRoute.js [hunk 5]
```diff
@@ -12,195 +12,183 @@ import Footer from "../components/modules/Footer";
...
+        <hr />
+        <h2>Spend points</h2>
+        Progress through our Ignition Program and earn points you can spend on
+        prizes!
+        <br />
+        <br />
+        <div className="collection collection--small-cards">
+          {// rendering our prizes
+          rewards.prizes.map((entry, i) => {
+            //return (<li><b>{entry.cost} Points:</b> {entry.description}</li>)
```
[the remark this answers is not in the document]
Comment (pyxld-kris) [reply src/routes/EarnRoute.js]: It certainly could, and now is! Commit 7fa2b8b64e3c1310ae6bce0931e4f23d5f62dc72

File: src/routes/PlayRoute.js (hunk 1 above)
Comment (pyxld-kris) [reply src/routes/PlayRoute.js]: Moved function body to useEffect and replaced function name with descriptive comment. Commit 1d6c1b00b1bc65c27e636e2213d14634446bb669

File: src/components/common/Button/Button.js [hunk 6]
```diff
@@ -6,29 +6,22 @@ import { toast } from "react-toastify";
...
-    super();
-    this.clickHandler = this.clickHandler.bind(this);
-  }
-
-  clickHandler() {
-    if (this.props.href) window.open(this.props.href, "_blank");
-    else if (this.props.toast) toast(this.props.toast);
+export default function Button(props) {
+  const clickHandler = () => {
+    if (props.href) window.open(props.href, "_blank");
```
[the remark this answers is not in the document]
Comment (pyxld-kris) [reply src/components/common/Button/Button.js]: You've caught a lot of my laziness! This is one I'll need to think about how best to resolve without having to restructure too many of our other files, and keep the simplicity of passing href in as a prop (unless this is bad practice?). The simplest option I see is to conditionally surround our buttons in <a> tags if the href prop is passed inside of render. 

Thoughts?

**Comment by pyxld-kris** [comment]
Resolved 4/6 of your comments. I left questions in two of them for you and will resolve those with a bit more info.

Will be creating some issues in the near future to collect opinions on overall structure and high level things relating to the project which I mentioned above. Hopefully will have something up this week!

**Pull request merged**